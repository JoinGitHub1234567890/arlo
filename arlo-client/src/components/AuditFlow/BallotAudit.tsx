import React from 'react'

interface Props {
  review: () => void
}

const BallotAudit: React.FC<Props> = ({ review }: Props) => {
  return <p>Auditing Ballot</p>
}

export default BallotAudit
