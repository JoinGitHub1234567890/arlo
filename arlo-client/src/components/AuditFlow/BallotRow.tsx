import styled from 'styled-components'

const BallotRow = styled.div`
  display: flex;
  justify-content: flex-start;
  margin: 20px 0;

  .ballot-side {
    width: 200px;
    padding: 20px 0;
  }
  .ballot-main {
    width: 50%;
    padding: 20px;
  }
`

export default BallotRow
