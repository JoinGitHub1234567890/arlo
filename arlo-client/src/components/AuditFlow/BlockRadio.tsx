import React from 'react'
import styled from 'styled-components'
import { IBallot } from '../../types'

const Block = styled.label`
  &.bp3-control.bp3-radio {
    display: inline-block;
    position: relative;
    margin-bottom: 20px;

    .bp3-control-indicator {
      border-radius: unset;
      width: 10em;
      height: 3em;

      &::before {
        display: none;
      }
      .radio-text {
        position: absolute;
        top: 50%;
        transform: translateY(-50%);
        width: 100%;
        padding: 5px;
        text-align: center;
      }
    }
  }
`

interface IProps {
  name?: string
  value: Exclude<IBallot['vote'], null>
  handleChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
  locked?: boolean
  className?: string
  checked?: boolean
}

const TEXT = {
  YES: 'Yes/For',
  NO: 'No/Against',
  NO_CONSENSUS: 'No audit board consensus',
  NO_VOTE: 'Blank vote/no mark',
}

const BlockRadio = ({
  name = '',
  value,
  handleChange,
  locked,
  className,
  checked,
}: IProps) => (
  <Block className={`${className} bp3-control bp3-radio`}>
    {locked ? (
      <input type="radio" name={name} value={value} disabled checked />
    ) : (
      <input
        type="radio"
        name={name}
        data-testid={value}
        value={value}
        onChange={handleChange}
        checked={checked}
      />
    )}
    <span className="bp3-control-indicator">
      <span className="radio-text">{TEXT[value]}</span>
    </span>
  </Block>
)

export default BlockRadio
