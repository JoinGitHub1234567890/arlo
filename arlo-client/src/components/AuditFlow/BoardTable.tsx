import React from 'react'
import styled from 'styled-components'
import { Table, Column, Cell } from '@blueprintjs/table'
import { H1, Button } from '@blueprintjs/core'
import { Link } from 'react-router-dom'
import { IAuditBoard, IBallot } from '../../types'

const RightWrapper = styled.div`
  display: flex;
  flex-direction: row-reverse;
  margin: 20px 0;
`

const PaddedCell = styled(Cell)`
  padding: 3px 5px;
`

const RightButton = styled(Button)`
  float: right;
`

const ActionWrapper = styled.div`
  margin-bottom: 20px;
  .bp3-checkbox {
    display: inline-block;
    margin-left: 20px;
  }
`

interface IProps {
  setIsLoading: (arg0: boolean) => void
  isLoading: boolean
  board: IAuditBoard
  url: string
}

const KEYS: ('tabulator' | 'batch' | 'position' | 'status')[] = [
  'tabulator',
  'batch',
  'position',
  'status',
]

const STATUSES: { [key: string]: string } = {
  AUDITED: 'Audited',
  NOT_AUDITED: 'Not Audited',
}

const BoardTable: React.FC<IProps> = ({ board, url }: IProps) => {
  const renderCell = (rI: number, cI: number) => {
    if (board.ballots) {
      const row: IBallot = board.ballots[rI]
      if (!KEYS[cI]) {
        return <PaddedCell>{board.name}</PaddedCell>
      } else if (STATUSES[row[KEYS[cI]]]) {
        const action =
          row[KEYS[cI]] === 'AUDITED' ? (
            <RightButton small>Re-audit</RightButton>
          ) : null
        return (
          <PaddedCell>
            <>
              {STATUSES[row[KEYS[cI]]]} {action}
            </>
          </PaddedCell>
        )
        // wrapping content in React.Fragment to avoid unexpected props being passed to dom elements and throwing warnings
      } else {
        return <PaddedCell>{row[KEYS[cI]]}</PaddedCell>
      }
    } else {
      return <PaddedCell loading />
    }
  }

  const columnWidths = (): (number | null)[] => {
    const container = document.getElementsByClassName(
      'board-table-container'
    )[0]
    if (!container) return Array(5).fill(null)
    const containerSize = container.clientWidth
    const colWidth = (containerSize - 31) / 5
    return Array(5).fill(colWidth)
  }

  const roundComplete =
    board.ballots && board.ballots.every(b => b.status === 'AUDITED')

  return (
    <div className="board-table-container">
      <H1>{board.name}: Ballot Cards to Audit</H1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
        commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
        velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
        occaecat cupidatat non proident, sunt in culpa qui officia deserunt
        mollit anim id est laborum.
      </p>
      <RightWrapper>
        {roundComplete ? (
          <Button intent="primary">Review Complete - Finish Round</Button>
        ) : (
          <Link
            to={url + '/round/1/ballot/1'}
            className="bp3-button bp3-intent-primary"
          >
            Start Auditing
          </Link>
        )}
      </RightWrapper>
      <ActionWrapper>
        {!roundComplete && (
          <>
            <Button intent="primary">Download Ballot List as CSV</Button>
          </>
        )}
      </ActionWrapper>
      <Table numRows={10} defaultRowHeight={30} columnWidths={columnWidths()}>
        <Column key="tabulator" name="Tabulator" cellRenderer={renderCell} />
        <Column key="batch" name="Batch" cellRenderer={renderCell} />
        <Column
          key="position"
          name="Record/Position"
          cellRenderer={renderCell}
        />
        <Column key="status" name="Status" cellRenderer={renderCell} />
        <Column name="Audit Board" cellRenderer={renderCell} />
      </Table>
    </div>
  )
}

export default BoardTable
