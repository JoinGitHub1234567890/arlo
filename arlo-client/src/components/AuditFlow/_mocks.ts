import { IAuditBoard } from '../../types'

export const dummyBoard: IAuditBoard[] = [
  {
    id: '123',
    name: 'Audit Board #1',
    members: [],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    ballots: [
      {
        tabulator: '11',
        batch: 'Precinct 13',
        position: '313',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
    ],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    ballots: [
      {
        tabulator: '11',
        batch: 'Precinct 13',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '17',
        batch: 'Precinct 19',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
      {
        tabulator: '23',
        batch: 'Precinct 29',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
      },
    ],
  },
]

export default {
  dummyBoard,
}
