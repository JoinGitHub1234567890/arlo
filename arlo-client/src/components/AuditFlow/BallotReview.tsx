import React from 'react'
import { H3, Divider, Button } from '@blueprintjs/core'
import styled from 'styled-components'
import { IReview } from '../../types'
import { BallotRow, FormBlock, ProgressActions } from './Atoms'
import FormButton from '../Form/FormButton'
import BlockRadio from './BlockRadio'

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 20px 0;
`

const SingleBlockRadio = styled(BlockRadio)`
  &.bp3-control.bp3-radio {
    margin: 0;
  }
`

interface IProps {
  contest: string
  goAudit: () => void
  review: IReview
}

const BallotReview: React.FC<IProps> = ({
  contest,
  goAudit,
  review: { vote, comment },
}: IProps) => {
  /* eslint-disable no-console */
  const handleSubmit = () => console.log(vote, comment)
  return (
    <BallotRow>
      <div className="ballot-side"></div>
      <div className="ballot-main">
        <FormBlock>
          <H3>{contest}</H3>
          <Divider />
          <Wrapper>
            <SingleBlockRadio value={vote} locked />
            <Button onClick={goAudit} icon="edit" minimal>
              Edit
            </Button>
          </Wrapper>
          <p>COMMENT: {comment}</p>
        </FormBlock>
        <ProgressActions>
          <FormButton type="submit" onClick={handleSubmit} intent="success">
            Submit &amp; Next Ballot
          </FormButton>
          <Button onClick={goAudit} minimal>
            Back
          </Button>
        </ProgressActions>
      </div>
    </BallotRow>
  )
}

export default BallotReview
