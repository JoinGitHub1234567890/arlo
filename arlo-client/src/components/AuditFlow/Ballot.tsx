import React, { useState } from 'react'
import { H1, H3, Callout, H4, Divider } from '@blueprintjs/core'
import styled from 'styled-components'
import BallotAudit from './BallotAudit'
import BallotReview from './BallotReview'
import { AuditBoard } from '../../types'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const MainCallout = styled(Callout)`
  width: 400px;
  font-weight: 700;
`

const Info = styled.div`
  display: flex;
  justify-content: flex-start;
  margin: 20px 0;

  .ballot-info {
    width: 200px;
    padding: 20px 0;
  }
  .description {
    width: 50%;
    padding: 20px;
  }
`

interface Props {
  roundId: string
  ballotId: string
  board: AuditBoard
}

const Ballot: React.FC<Props> = ({ roundId, ballotId, board }: Props) => {
  const [auditing, setAuditing] = useState(true)

  if (!board.ballots) return null // TODO handle informatively
  const ballot = board.ballots[Number(ballotId) - 1]
  if (!ballot) return null // TODO handle informatively

  return (
    <Wrapper>
      <H1>{board.name}: Ballot Card Data Entry</H1>
      <H3>Enter Ballot Information</H3>
      <MainCallout icon={null} intent="primary">
        Round {roundId}: auditing ballot {ballotId} of {board.ballots.length}
      </MainCallout>
      <Info>
        <div className="ballot-info">
          <H4>Current ballot:</H4>
          <div>Tabulator: {ballot.tabulator}</div>
          <div>Batch: {ballot.batch}</div>
          <div>Record/Position: {ballot.record}</div>
        </div>
        <Divider />
        <div className="description">
          <H4>Are you looking at the correct ballot?</H4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
            ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in
            reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
            pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
            culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </Info>
      {auditing ? (
        <BallotAudit review={() => setAuditing(false)} />
      ) : (
        <BallotReview audit={() => setAuditing(true)} />
      )}
    </Wrapper>
  )
}

export default Ballot
