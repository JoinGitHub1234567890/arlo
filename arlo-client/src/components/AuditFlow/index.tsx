import React, { useState, useEffect, useCallback } from 'react'
import { H1 } from '@blueprintjs/core'
import { Route, Switch } from 'react-router-dom'
import { IAuditFlowParams, IAudit, IAuditBoard } from '../../types'
import { api } from '../utilities'
import { statusStates } from '../AuditForms/_mocks'
import BoardTable from './BoardTable'
import MemberForm from './MemberForm'
import Ballot from './Ballot'
import Wrapper from '../Atoms/Wrapper'

const rand = (max: number = 100, min: number = 1) =>
  Math.floor(Math.random() * (+max - +min)) + +min

const dummyBoard: IAuditBoard[] = [
  {
    id: '123',
    name: 'Audit Board #1',
    members: [],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    ballots: Array(10)
      .fill('')
      .map(() => ({
        tabulator: '' + rand(),
        batch: `Precinct ${rand()}`,
        position: '' + rand(2000),
        status: ['AUDITED', 'NOT_AUDITED'][rand(2, 0)] as
          | 'AUDITED'
          | 'NOT_AUDITED',
        vote: null,
        comment: '',
      })),
  },
]

interface IProps {
  match: {
    params: IAuditFlowParams
    url: string
  }
}

const AuditFlow: React.FC<IProps> = ({
  match: {
    params: { electionId, token },
    url,
  },
}: IProps) => {
  const [isLoading, setIsLoading] = useState<boolean>(false)

  const [audit, setAudit] = useState(statusStates[3])

  const getStatus = useCallback(async (): Promise<IAudit> => {
    const audit: IAudit = await api('/audit/status', { electionId })
    return audit
  }, [electionId])

  const updateAudit = useCallback(async () => {
    const audit = await getStatus()
    setIsLoading(true)
    setAudit(audit)
    setIsLoading(false)
  }, [getStatus])

  useEffect(() => {
    updateAudit()
  }, [updateAudit])

  const [dummy, setDummy] = useState(2)
  const board = {
    ...audit.jurisdictions[0].auditBoards.find(
      (v: IAuditBoard) => v.id === token
    ),
    ...dummyBoard[dummy],
  }

  if (!board) {
    return (
      <Wrapper>
        <H1>
          Sorry, but that Audit Board does not exist in{' '}
          {audit.jurisdictions[0].name} and {audit.name}
        </H1>
      </Wrapper>
    )
  } else if (board.members.length) {
    return (
      <Wrapper>
        <Switch>
          <Route
            exact
            path="/election/:electionId/board/:token"
            render={({ match: { url } }) => (
              <BoardTable
                isLoading={isLoading}
                setIsLoading={setIsLoading}
                board={board}
                url={url}
              />
            )}
          />
          <Route
            path={url + '/round/:roundId/ballot/:ballotId'}
            render={({
              match: {
                params: { roundId, ballotId },
              },
            }) => (
              <Ballot roundId={roundId} ballotId={ballotId} board={board} />
            )}
          />
        </Switch>
      </Wrapper>
    )
  } else {
    return (
      <Wrapper>
        <MemberForm
          isLoading={isLoading}
          setIsLoading={setIsLoading}
          setDummy={setDummy}
          boardName={board.name}
          jurisdictionName={audit.jurisdictions[0].name}
        />
      </Wrapper>
    )
  }
}

export default AuditFlow
