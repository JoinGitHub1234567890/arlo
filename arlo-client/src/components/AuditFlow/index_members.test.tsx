import React from 'react'
import { render, wait } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { routerTestProps } from '../testUtilities'
import AuditFlow from './index'
import { dummyBoard, dummyBallots } from './_mocks'
import statusStates from '../AuditForms/_mocks'
import { api } from '../utilities'
import { IAudit, IBallot } from '../../types'

const memberlessDummy = statusStates[3]

const apiMock = api as jest.Mock<ReturnType<typeof api>, Parameters<typeof api>>

jest.mock('../utilities')

const memberingMock = async (
  endpoint: string
): Promise<IAudit | { ballots: IBallot[] }> => {
  switch (endpoint) {
    case '/audit/status':
      memberlessDummy.jurisdictions[0].auditBoards = [dummyBoard[0]]
      return memberlessDummy
    default:
      return dummyBallots
  }
}

afterEach(() => {
  apiMock.mockClear()
})

const routeProps = routerTestProps('/election/:electionId/board/:token', {
  electionId: '1',
  token: '123',
})

describe('AuditFlow initial load', () => {
  beforeEach(() => {
    apiMock.mockImplementation(memberingMock)
  })

  it('renders correctly', () => {
    const { container } = render(
      <Router {...routeProps}>
        <AuditFlow {...routeProps} />
      </Router>
    )
    expect(container).toMatchSnapshot()
  })

  it('fetches initial state from api', async () => {
    const { container } = render(
      <Router {...routeProps}>
        <AuditFlow {...routeProps} />
      </Router>
    )
    await wait(() => {
      expect(apiMock).toBeCalled()
      expect(container).toMatchSnapshot()
    })
  })

  it('renders member form', async () => {
    const { container, getByText } = render(
      <Router {...routeProps}>
        <AuditFlow {...routeProps} />
      </Router>
    )
    await wait(() => {
      expect(apiMock).toBeCalled()
      expect(
        getByText('Member Sign in for Audit Board: Audit Board #1')
      ).toBeTruthy()
      expect(container).toMatchSnapshot()
    })
  })
})
