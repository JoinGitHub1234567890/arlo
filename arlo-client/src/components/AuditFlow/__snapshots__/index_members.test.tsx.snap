// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditFlow initial load fetches initial state from api 1`] = `
<div>
  <div
    class="sc-eNQAEJ jDJHSN"
  >
    <h1
      class="bp3-heading"
    >
      Member Sign in for Audit Board: 
      Audit Board #1
    </h1>
    <p>
      Enter the information below for each member of 
      Jurisdiction 1
       
      Audit Board #1
       below, then click "Next" to proceed.
    </p>
    <div
      class="sc-bxivhb cfvxmq"
    >
      <h2
        class="bp3-heading sc-htpNat fysvnw"
      />
      <form>
        <div
          class="sc-ifAKCX iKQdjg"
        >
          <h3
            class="bp3-heading sc-bZQynM vxpym"
          >
            Audit Board Member
          </h3>
          <label
            for="[0]name"
          >
            <span
              class="sc-htoDjs flCEIS"
            >
              Full Name
            </span>
            <input
              class="sc-dnqmqq jyHOFs"
              id="[0]name"
              name="[0]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-htoDjs flCEIS"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[0]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <div
          class="sc-ifAKCX iKQdjg"
        >
          <h3
            class="bp3-heading sc-bZQynM vxpym"
          >
            Audit Board Member
          </h3>
          <label
            for="[1]name"
          >
            <span
              class="sc-htoDjs flCEIS"
            >
              Full Name
            </span>
            <input
              class="sc-dnqmqq jyHOFs"
              id="[1]name"
              name="[1]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-htoDjs flCEIS"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[1]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <button
          class="bp3-button bp3-intent-primary sc-gzVnrw eLDrzZ"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Next
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`AuditFlow initial load renders correctly 1`] = `
<div>
  <div
    class="sc-eNQAEJ jDJHSN"
  >
    <div
      class="bp3-spinner"
    >
      <div
        class="bp3-spinner-animation"
      >
        <svg
          height="50"
          stroke-width="8.00"
          viewBox="1.00 1.00 98.00 98.00"
          width="50"
        >
          <path
            class="bp3-spinner-track"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
          />
          <path
            class="bp3-spinner-head"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
            pathLength="280"
            stroke-dasharray="280 280"
            stroke-dashoffset="210"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`AuditFlow initial load renders member form 1`] = `
<div>
  <div
    class="sc-eNQAEJ jDJHSN"
  >
    <h1
      class="bp3-heading"
    >
      Member Sign in for Audit Board: 
      Audit Board #1
    </h1>
    <p>
      Enter the information below for each member of 
      Jurisdiction 1
       
      Audit Board #1
       below, then click "Next" to proceed.
    </p>
    <div
      class="sc-bxivhb cfvxmq"
    >
      <h2
        class="bp3-heading sc-htpNat fysvnw"
      />
      <form>
        <div
          class="sc-ifAKCX iKQdjg"
        >
          <h3
            class="bp3-heading sc-bZQynM vxpym"
          >
            Audit Board Member
          </h3>
          <label
            for="[0]name"
          >
            <span
              class="sc-htoDjs flCEIS"
            >
              Full Name
            </span>
            <input
              class="sc-dnqmqq jyHOFs"
              id="[0]name"
              name="[0]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-htoDjs flCEIS"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[0]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <div
          class="sc-ifAKCX iKQdjg"
        >
          <h3
            class="bp3-heading sc-bZQynM vxpym"
          >
            Audit Board Member
          </h3>
          <label
            for="[1]name"
          >
            <span
              class="sc-htoDjs flCEIS"
            >
              Full Name
            </span>
            <input
              class="sc-dnqmqq jyHOFs"
              id="[1]name"
              name="[1]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-htoDjs flCEIS"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[1]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <button
          class="bp3-button bp3-intent-primary sc-gzVnrw eLDrzZ"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Next
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
